### PR TITLE
feat: IBC transfer acknowledgement implemented

### PR DESCRIPTION
**Pull request by swelf19**
implements https://p2pvalidator.atlassian.net/jira/software/projects/LSC/boards/121?selectedIssue=LSC-87

Implemented a wrapper for original ibc transfer module witch handles an ack packet and call a sudo handler with appropriate payload.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -128,7 +128,6 @@ require (
 )
 
 replace (
-	// github.com/CosmWasm/wasmd v0.27.0 => ../wasmd
 	github.com/CosmWasm/wasmd v0.27.0 => github.com/swelf19/wasmd v0.27.1-0.20220705054332-4f6cf43be829
```
Comment (ratik) [inline go.mod]: maybe move this one to neutron org?

File: x/sudo/sudo.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+}
+
+func NewSudoHandler(wasmKeeper *wasm.Keeper, moduleName string) SudoHandler {
+	return SudoHandler{
+		moduleName: moduleName,
+		wasmKeeper: wasmKeeper,
+	}
+}
+
+// Sudo allows privileged access to a contract. This can never be called by an external tx, but only by
```
Comment (zavgorodnii) [inline x/sudo/sudo.go]: don't think this comment should be here

File: x/sudo/sudo.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+//
+func (s *SudoHandler) SudoResponse(
+	ctx sdk.Context,
+	contractAddress sdk.AccAddress,
+	request channeltypes.Packet,
+	msg []byte,
+) ([]byte, error) {
+	s.Logger(ctx).Info("SudoResponse", "contractAddress", contractAddress, "request", request, "msg", msg)
+
+	// basically just for unit tests right now. But i think we will have the same logic in the production
```
Comment (zavgorodnii) [inline x/sudo/sudo.go]: please make this a todo so that we can find this place easily during final clean up

File: x/sudo/sudo.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+}
+
+func (s *SudoHandler) SudoTimeout(
+	ctx sdk.Context,
+	contractAddress sdk.AccAddress,
+	request channeltypes.Packet,
+) ([]byte, error) {
+	s.Logger(ctx).Info("SudoTimeout", "contractAddress", contractAddress, "request", request)
+
+	// basically just for unit tests right now. But i think we will have the same logic in the production
```
Comment (zavgorodnii) [inline x/sudo/sudo.go]: same here

File: x/transfer/module.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+// OnAcknowledgementPacket implements the IBCModule interface.
+func (im IBCModule) OnAcknowledgementPacket(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	acknowledgement []byte,
+	relayer sdk.AccAddress,
+) error {
+	err := im.IBCModule.OnAcknowledgementPacket(ctx, packet, acknowledgement, relayer)
```
Comment (zavgorodnii) [inline x/transfer/module.go:42]: maybe we should make it clear which handler returned the error here? (e.g., wrap the error)

**Comment by ratik** [comment]
can we have a bit of description here and link to the task?

**Review by zavgorodnii — APPROVED** [review]

**Review by pr0n00gler — COMMENTED** [review]
I don't really like the idea of putting `sudo` and `transfer` in `/x` folder. That is a place for full Cosmos-SDK modules (with keepers, msg-handlers, etc). In our case `sudo` and `transfer` are just helpers for our `interchaintxs` module.

**Review by pr0n00gler — COMMENTED** [review]
I may be wrong about `transfer` because it's a wrapper around the IBC transfer module, but `sudo` definitely should be moved to another place

**Review by pr0n00gler — COMMENTED** [review]
Could you add more comments for the `transfer` module?
Because it just sits in `/x` folder, but it's not clear for strangers at all what is it, why we need this, what it does, etc.

Would be good if `sudo` has comment coverage too. Basic at least

**Review by pr0n00gler — APPROVED** [review]

**Pull request merged**